rest/data/task.go

```diff
@@ -242,8 +242,8 @@ func (tc *DBTaskConnector) GetManifestByTask(taskId string) (*manifest.Manifest,
 
 // FindTasksByVersion gets all tasks for a specific version
 // Results can be filtered by task name, variant name and status in addition to being paginated and limited
-func (tc *DBTaskConnector) FindTasksByVersion(versionID string, statuses []string, baseStatuses []string, variant string, taskName string, page, limit int, fieldsToProject []string, sorts []task.TasksSortOrder) ([]task.Task, int, error) {
-	tasks, total, err := task.GetTasksByVersion(versionID, sorts, statuses, baseStatuses, variant, taskName, page, limit, fieldsToProject)
+func (tc *DBTaskConnector) FindTasksByVersion(versionID string, statuses []string, baseStatuses []string, variants []string, taskNames []string, page, limit int, fieldsToProject []string, sorts []task.TasksSortOrder) ([]task.Task, int, error) {
```

Comment:
Sorry to ask this of you because it probably should've been done before, but this is getting really unwieldy; having this many inputs of the same type so close together introduces a lot of room for programmer error, and makes code very difficult to read. Would you mind refactoring this to take in a filterOptions struct or something, that contains the fields between statuses -> sort? (This also means you should just be able to pass in a blank filterOptions struct when you don't want to filter, which would be nice)